### DISCUSSION TITLE
Exclude App from Ice?

### DISCUSSION THREAD

**Discussion by brijazz** [jordanbaird/Ice#101] [General]
Greetings from a Bartender refugee! It looks like [Doll.app](https://github.com/xiaogdgenuine/Doll) doesn't want to play nicely with Ice (every time it generates a new notification, Ice puts it into its 'always hidden' section). Is there a way to tell Ice to ignore this app?

**Comment by jordanbaird** [comment]
Unfortunately not at the moment. The reason this is happening is that the always hidden section is at the end of the menu bar, which is where macOS automatically puts new menu bar items. I plan on adding a feature where you can specify where you want new items to appear. Once this gets implemented, I think it should solve your problem.

**Reply by joaofrgomes** [reply]
I was about to comment this as an issue, which I've identified with both Dropbox and a smart card reader plugin from our government administration…

As much as I've tried to drag them to different spots, it seems Ice doesn't keep or enforce a database with menu extra positions, am I right? That kind of precludes me from switching from my perpetual Bartender 5 license, especially on the Air but probably on all machines because having to move stuff around every time I reboot or log out is a bit of a hassle, but as soon as that's fixed, I should be good. Hopefully before the next major macOS update, v.15, hits?

Anyway, if and when that's the case (along with fixes to the whole notch situation), a donation is also in order. It's sad that I bought that awesome but now beleaguered piece of software back in November, but it was a long time coming, and my need for this kind of solution won't go away any time soon. Heck, even with no fixes for the notch, I might keep an outdated but still functional Bartender 5.0.48 on the Air, and move my other machines to Ice ASAP.

As for other stuff, like the GUI doohickey for custom/predetermined spacing and padding values that's obviously in the roadmap, it'll obviously be a welcome addition but I don't even care much for it when it comes to switching to Ice, [as those can be easily fixed via Terminal commands already](https://apple.stackexchange.com/questions/406316/can-the-spacing-of-menu-bar-apps-be-modified-in-macos-big-sur-and-later) – macOS Catalina's/Bartender's nice default compact values being "-int 6" for each, despite recommendations on said post to use twice as much padding as the spacing.

**Comment by brijazz** [comment]
Ah, makes sense. I'll watch for updates and the eventual rollout of this feature - thanks!

**Comment by ksignorini** [comment]
Another app like this is Badgeify. Badgeify is fantastic for adding apps that normally don't have menu extra positions, up there in the menu bar. Without ICE remembering where to put it, it keeps getting added to always hidden/hidden. That makes Badgeify with ICE totally useless.

Please add something to fix this. 

Thanks.

**Comment by dgreenberg19** [comment]
I know you're busy with Tahoe, but just want to add another vote for getting Doll to work when you have the opportunity.

**Comment by brijazz** [comment]
Any movement on the addition of this feature? I'm on Ice 0.11.12 and am still experiencing the issue described here.

**Comment by DT9** [comment]
Bump. Netbird (best tailscale / zerotier alternative) breaks too.
